tests/unit/test_charm.py

```diff
@@ -80,14 +80,42 @@ def test_given_private_key_is_stored_and_certificate_relation_is_created_when_on
             },
         )
         patch_generate_csr.return_value = csr.encode()
+        self.harness._backend.secret_add(label="csr", content={"csr": csr})
         self.harness.add_relation(relation_name="certificates", remote_app="certificates-provider")
 
         self.harness.update_config(key_values={"subject": "banana.com"})
 
         patch_request_certificate.assert_called_with(certificate_signing_request=csr.encode())
 
+    @patch(
+        "charms.tls_certificates_interface.v2.tls_certificates.TLSCertificatesRequiresV2.request_certificate_creation"  # noqa: E501, W505
+    )
+    @patch("charm.generate_csr")
+    def test_given_private_key_is_stored_and_csr_is_not_stored_certificate_relation_is_created_when_on_config_changed_then_certificate_is_not_requested(  # noqa: E501
+        self,
+        patch_generate_csr,
+        patch_request_certificate,
+    ):
+        private_key = "whatever private key"
+        private_key_password = "whatever password"
+        csr = "whatever csr"
+        self.harness.set_leader(is_leader=True)
+        self.harness._backend.secret_add(
+            label="private-key",
+            content={
+                "private-key": private_key,
+                "private-key-password": private_key_password,
+            },
+        )
+        patch_generate_csr.return_value = csr.encode()
+        self.harness.add_relation(relation_name="certificates", remote_app="certificates-provider")
+
+        self.harness.update_config(key_values={"subject": "banana.com"})
+
+        patch_request_certificate.assert_not_called()
+
     @patch("charm.generate_csr")
-    def test_given_certificate_is_requested_when_on_config_changed_then_status_is_waiting(
+    def test_given_certificate_is_requested_and_csr_is_not_stored_when_on_config_changed_then_status_is_active(  # noqa: E501
```

Comment:
I don't think we can get to this status anymore.
We have sorted out the path to request a certificate without having a csr stored.

Reply:
I removed this test.